content/5_Plat/RURQ.mdx

```diff
@@ -115,11 +133,159 @@ int main() {
 
 <Problems problems="bitProb" />
 
-## Lazy Segment Tree
+# Lazy Segment Tree
+
+Lazy segment trees allow us to efficiently perform range updates and range queries.
+
+<Resources>
+	<Resource
+		source="CF EDU"
+		title="Segment Tree Pt 2"
+		url="https://codeforces.com/edu/course/2/lesson/5"
+		starred
+	 />
+	<Resource source="CPH" title="28.1 - Segment Trees Revisited" starred>
+		short description
+	</Resource>
+	<Resource source="CSA" title="Segment Trees" url="segment_trees" starred>
+		interactive
+	</Resource>
+	<Resource
+		source="cp-algo"
+		title="Segment Tree"
+		url="data_structures/segment_tree.html#range-updates-lazy-propagation"
+		starred
+	>
+		adding on segments, assigning
+	</Resource>
+	<Resource source="CF" title="Efficient and easy segment trees" url="18051">
+		code is more confusing than recursive version
+	</Resource>
+</Resources>
+
+Lazy segment trees allow us to perform range updates and range queries in
+$\mathcal{O}(\log{N})$ time. In order to perform range updates in
+$\mathcal{O}(\log{N})$, we lazily apply updates to the nodes. That is, we store
+updates on the nodes that compose the range we are updating, and lazily apply
+the updates when we walk down the tree.
+
+As an example, let's consider writing a lazy segment tree that supports range
+addition and range summation. We would have our tree nodes contain the sum on
+the range, and keep a separate array keeping track of the lazy additions we
+have to do for each node. If we were to add $3$ to our array $a[0 \ldots n - 1]$,
+then we would need to set a lazy update at the root of our tree indicating that
+we need to add $3$ to the entire range. Then, if we were to query a subarray of
+$a$, we need to "push down" the updates at the root down to its children.
+
+It is crucial that we always push down all the previous updates when traversing
+down the tree. If we don't, the following things can happen:
+
+1. If we have two updates that affect two nodes, where one node is an ancestor of another, then one of the updates will be ignored.
+2. The updates will be applied in incorrect order, which matters if the updates are not commutative.
+
+For a concrete example for why point 1 can be an issue, consider the following
+sequence of updates.
+
+1. +3 to all values on $[1, 4]$
+2. +5 to all values on $[1, 2]$
+3. Query sum on $[1, 8]$
+
+If we don't push down the updates when traversing down the tree, then the first
+update will get ignored in our sum query.
+
+Here's an implementation of the previous focus problem, Horrible Queries.
```

Comment:
maybe add header and time comp again here for navigation purposes

Reply:
done lmk if theres any issues w/ it